pkg/termcorder/termcorder.go

```diff
@@ -0,0 +1,225 @@
+	return tc, nil
+}
+
+//TermcordingFromFlags parses command line flags (and arguments) and returns a pointer to a
+//new variable of type `Termcording`.
+func TermcordingFromFlags(options ...func(*Termcording) error) (*Termcording, error) {
+	var fName, cmdName string
+	var cmdArgs []string
+
+	var h, q, a, k bool
+
+	cli = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	cli.BoolVar(&h, "h", false, "Prints this message")
+	cli.BoolVar(&q, "q", false, "Quiet mode - suppresses the recording start and end prompts")
+	cli.BoolVar(&a, "a", false, "Appends to file instead of overwriting it")
+	cli.BoolVar(&k, "k", false, "Log key strokes to file as well")
+
+	cli.Parse(os.Args[1:])
+
+	shell, ok := os.LookupEnv("SHELL")
+	if cli.Arg(1) == "" && (!ok || shell == "") {
+		return &Termcording{}, errors.New("shell not set")
+	}
+	interactive := true
+	switch cli.NArg() {
+	case 0:
+		fName = defaultFileName
+		cmdName = shell
+	case 1:
+		fName = cli.Arg(0)
+		cmdName = shell
+	default:
+		fName = cli.Arg(0)
+		cmdName = cli.Arg(1)
+		cmdArgs = cli.Args()[2:]
+		interactive = false
+	}
+
+	return NewTermcording(&Config{
+		Filename:      fName,
+		CmdName:       cmdName,
+		CmdArgs:       cmdArgs,
+		QuietMode:     q,
+		Append:        a,
+		Interactive:   interactive,
+		LogKeystrokes: k,
+		PrintHelp:     h,
+	}, options...)
+}
+
+//Start runs the tc.cmd in a pseudo-terminal and writes all output to tc.out
+func (tc *Termcording) Start() error {
+	if tc.Config.PrintHelp {
+		if tc.out == nil {
+			tc.out = os.Stdout
+		}
+		printHelp(tc)
+		return nil
+	}
+
+	if tc.cmd == nil {
+		tc.cmd = exec.Command(tc.Config.CmdName, tc.Config.CmdArgs...)
+	}
+
+	if !tc.Config.QuietMode {
+		fmt.Println("Starting recording session. CTRL-D to end.")
+		defer fmt.Printf("\nRecording session ended. Session saved to %s\n", tc.Config.Filename)
+	}
+
+	pterm, restoreMode, err := pseudoTermFromCmd(tc.cmd, tc.Config.Interactive)
+	if err != nil {
+		return fmt.Errorf("failed to create a pty from command: %s", err)
+	}
+	defer func() {
+		restoreMode()
+		pterm.Close()
+	}()
+
+	tc.in = io.Writer(pterm)
+	if tc.out == nil {
+		closerFn, err := tc.setupWriters()
+		if err != nil {
+			return fmt.Errorf("failed to set up recording outputs: %s", err)
+		}
+		defer closerFn()
+	}
+
+	go func() {
+		io.Copy(tc.in, os.Stdin)
+	}()
+
+	io.Copy(tc.out, pterm)
+
+	return tc.cmd.Wait()
+}
+
+func (tc *Termcording) setupWriters() (closer func() error, err error) {
+	mode := os.O_TRUNC
+	if tc.Config.Append {
+		mode = os.O_APPEND
+	}
+	f, err := os.OpenFile(tc.Config.Filename, os.O_WRONLY|os.O_CREATE|mode, 0700)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open recording file: %s", err)
+	}
+	if tc.Config.LogKeystrokes {
+		tc.in = io.MultiWriter(tc.in, f)
+	}
+	tc.out = io.MultiWriter(os.Stdout, f)
+	return f.Close, nil
+}
+
+func pseudoTermFromCmd(c *exec.Cmd, interactive bool) (pterm *os.File, stdinModeRestore func(), err error) {
+	var s *pty.Winsize
+	if interactive {
```

Comment:
`pty.GersizeFull` returns an error if the command we're running is not interactive (e.g. not a shell) so we'll need to add another `if interactive` check here. Not sure I prefer this over manually sending the signal to `ch`. Any thoughts/recommendations?